projects/ui/src/pages/l1transfer.tsx

```diff
@@ -0,0 +1,649 @@
+                  <Warning>
+                    {`Transfer amount higher than your ${copy.MODES[values.fromMode]}.`}
+                  </Warning>
+                )
+              )}
+              {toMode === FarmToMode.INTERNAL && !ethTransferCheck && (
+                <Warning color="error">
+                  If you send assets to the Farm Balance of contracts, centralized
+                  exchanges, etc. that don&apos;t support Farm Balances, the assets
+                  will be lost.
+                </Warning>
+              )}
+              <Button
+                type="submit"
+                variant="contained"
+                color="primary"
+                size="large"
+                disabled={!isValid || isSubmitting}
+              >
+                {noBalance
+                  ? 'Nothing to transfer'
+                  : values.approving ? `Approve ${tokenIn.symbol}` // : !enoughBalanceCheck
+                    // ? 'Not enough to transfer'
+                    : 'Transfer'}
+              </Button>
+            </Stack>
+          </Form>
+        </Card>
+      </Box>
+    );
+  };
+
+// ---------------------------------------------------
+
+const SUPPORTED_TOKENS = [
+  WETH,
+  WSTETH,
+  DAI,
+  USDC,
+  USDT,
+];
+
+const L1Transfer: FC<{}> = () => {
+  /// Ledger
+  const account = useAccount();
+  const { data: signer } = useSigner();
+  const beanstalk = useBeanstalkContract(signer);
+
+  /// Tokens
+  const getChainToken = useGetChainToken();
+  const weth = getChainToken(WETH);
+
+  /// Token List
+  const tokenMap = useTokenMap<ERC20Token | NativeToken>(SUPPORTED_TOKENS);
+  const tokenList = useMemo(() => Object.values(tokenMap), [tokenMap]);
+
+  /// Form
+  const middleware = useFormMiddleware();
+  const initialValues: TransferFormValues = useMemo(
+    () => ({
+      tokensIn: [
+        {
+          token: weth,
+          amount: undefined,
+        },
+      ],
+      balanceFrom: BalanceFrom.INTERNAL,
+      fromMode: FarmFromMode.INTERNAL,
+      toMode: FarmToMode.EXTERNAL,
+      destination: '',
+      approving: false,
+    }),
+    [weth]
+  );
+
+  const sdk = useSdk();
+  const beanstalkL1Address = sdk.addresses.BEANSTALK.ETH_MAINNET;
+  const beanstalkL1 = new ethers.Contract(beanstalkL1Address, beanstalkL1MiniAbi, signer)
+
+  const supportedAddresses = Object.keys(tokenMap);
+
+  const tokenBalances = useReadContract({
+    address: beanstalkL1Address as `0x${string}`,
+    abi: JSON.parse(beanstalkL1MiniAbi.format(ethers.utils.FormatTypes.json) as string),
+    functionName: "getAllBalances",
+    // @ts-ignore
+    args: [account, supportedAddresses],
+    query: {
+      enabled: Boolean(account),
+      refetchInterval: 10000
+    }
+  });
+
+  const tokenBalanceData = tokenBalances.data as Array<{ internal: bigint, external: bigint, total: bigint }>;
+  const tokenData = supportedAddresses.reduce((acc, address, index) => {
+    // @ts-ignore
```

Comment:
any way to not have the ts-ignore ? and below

Reply:
Removed!